### PR TITLE
chore: de-duplicate the zoe contract types and make them more organized

### LINKED ISSUE #2152: deduplicate type declarations in loan contract
**What is the Problem Being Solved?**

Some types are declared in both `packages/zoe/src/contracts/loan/types.js` and `packages/zoe/src/contracts/exported.js`. They should only appear in one place. I assume it should be in the types file, and `exported.js` should import and re-export a subset of types.

BorrowerConfigProperties
LenderSeatProperty
LoanTerms
AutoswapInstance
LoanConfigWithBorrower
LoanConfigWithLender
PeriodNotifier
BorrowFacet
DebtCalculatorConfig
MMR
InterestRate

**Description of the Design**

--

**Security Considerations**

None.

**Test Plan**

If the typescript checker remains happy, it'll be fine.

### PR DESCRIPTION
**Pull request by katelynsills**
This PR organizes the types files in zoe contracts. We had duplicate types for the loan contract. I think we should probably not define types in exported.js as a general rule, but rather define them elsewhere and import them in exported.js.

If possible, let's get this merged before https://github.com/Agoric/agoric-sdk/pull/2144 so I can better review the type changes there.

Closes #2152

### CODE REVIEW

**Review by michaelfig — COMMENTED** [review]
Looks good.  The ultimate test is to try munging one of the types in each of these files and ensuring that the eslint step fails as expected.  Not sure if you already did that.

**Comment by katelynsills** [comment]
> Looks good. The ultimate test is to try munging one of the types in each of these files and ensuring that the eslint step fails as expected. Not sure if you already did that.

I moused over and things were defined. I also just tried the munging of the types and that also worked

File: packages/zoe/src/contracts/loan/types.js [hunk 1]
```diff
@@ -76,12 +76,31 @@
...
+
 /**
  * @typedef {LoanConfigWithLender & BorrowerConfigProperties } LoanConfigWithBorrower
  *
  * The loan has a lender, a borrower, and collateral escrowed.
  */
 
+/**
+ * @typedef {LoanConfigWithLender & BorrowerConfigPropertiesMinusDebt
+ * } LoanConfigWithBorrowerMinusDebt
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contracts/loan/types.js:101]: It's unusual to see typedefs broken with a curly starting the line. When it's less than ~100 characters, I prefer the declared name at the end of the line.  

Your choice here, as I don't think we have standard formatting rules

File: packages/zoe/src/contracts/priceAggregatorTypes.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ * @property {(query: any) => ERef<any>} query make an unpaid query
+ */
+
+/**
+ * @typedef {Object} OracleCreatorFacet the private methods accessible from the
+ * contract instance
+ * @property {() => AmountKeywordRecord} getCurrentFees get the current
+ * fee amounts
+ * @property {(total: boolean=) => ERef<Invitation>}
+ * makeWithdrawInvitation create an invitation to withdraw fees
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contracts/priceAggregatorTypes.js:39]: I prefer the declared name on the right end of the line after the curly. It's hard enough to visually scan these with their ragged placement, but after a curly stands out a little in the IDE.  If we consistently had the declared type left flush, that would be a great consistency, but when they're mostly on the right, I find these exceptions to be hard to find.

**Review by Chris-Hibbert — APPROVED** [review]
These seem like sensible groupings. Thanks for taking the time.  

A few quibbles about formatting, but if you don't agree, I'm completely fine with your judgement. The main target for these is the types that will be displayed by the IDE, not reading them directly.

File: packages/zoe/src/contracts/loan/types.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/src/contracts/loan/types.js:101]: Ah, this was an automatic change from my linters. Let me see if I can undo it.

Comment (katelynsills) [reply packages/zoe/src/contracts/loan/types.js:101]: Ah, actually, I just redid this, and this particular formatting was already there. I didn't change it in this PR, so I'm going to punt any fixing to a different PR if desired

File: packages/zoe/src/contracts/priceAggregatorTypes.js (hunk 2 above)
Comment (katelynsills) [reply packages/zoe/src/contracts/priceAggregatorTypes.js:39]: This was also not a change made in this PR

**Pull request merged**